screens/RegisterScreen.tsx

```diff
@@ -73,13 +81,13 @@ const RegisterScreen = ({ navigation }: Props) => {
               style={authStyles.button}
               icon="login"
               mode="contained"
-              onPress={() => submit()}
+              onPress={() => handleRegister()}
             >
-              Sign Up
+              {loading ? 'Loggin in' : 'Register'}
             </Button>
             <View style={authStyles.linkTextContainer}>
               <Text>Already have an account? </Text>
-              <TouchableOpacity onPress={() => navigation.navigate('Login')}>
+              <TouchableOpacity onPress={() => handleNavigate()}>
```

Comment:
No need to create a "new" function since it doesn't take any parameters and already exists, so just gotta reference it
```suggestion
              <TouchableOpacity onPress={handleNavigate}>
```